bindings/rust/.clippy.toml

```diff
@@ -0,0 +1,2 @@
+# This should match rust-toolchain
+msrv = "1.57.0"
```

Comment:
It would be nice if this could automatically be consumed from rust-toolchain, but I can't think of a nice way to do that.

I was hoping we could get by with sym-linking our `.clippy.toml` to `rust-toolchain.toml`, but looks like `rust-toolchain.toml` has a different format.

Reply:
Yeah, and we're not even using rust-toolchain.toml, we're using the old legacy ascii format. So I couldn't even put a comment in there reminding people to update clippy.

Reply:
We might be able to add a manual check / comparison to our CI, but that'd be a bit hacky

Reply:
Could we switch our `rust-toolchain` to `rust-toolchainl.toml`. That way we'd be able to add a comment? 

Though the version should still show up in a "find and replace" with the current structure, so I'm not too worried about it.

Reply:
https://github.com/aws/s2n-tls/issues/3786